File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> seen; 
        int max_count = 0; 

        for(int i = 0; i < nums.size(); ++i) {
            if(!seen.count(nums[i])) {
                seen.insert(nums[i]); 
            } 
        }

        for(int i = 0; i < nums.size(); ++i) {
            if(seen.count(nums[i]-1)) continue; 

            int count = 0; 
            int curNum = nums[i]; 
            while(seen.count(curNum)) {
                ++count; 
                ++curNum; 
            }

            max_count = max(count, max_count); 
        }

        return max_count; 
    }
};

File: Data Structures & Algorithms/palindromic-substrings/submission-0.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        seen = [[1] * len(s)]

        count = len(s)

        for i in range(1, len(s)): 
            results = []

            for j in range(len(s) - i): 
                if i == 1: 
                    if s[j] == s[j+i]: 
                        count += 1
                        results.append(1)
                    else: 
                        results.append(0)
                else: 
                    if s[j] == s[j+i] and seen[i-2][j+1]: 
                        count += 1
                        results.append(1)
                    else: 
                        results.append(0)
            
            seen.append(results)
        
        return count